modules/cas_cache/layer_cache_management.c

```diff
@@ -585,8 +585,9 @@ static void _cache_mngt_cache_stop_complete(ocf_cache_t cache, void *priv,
 {
 	struct _cache_mngt_stop_context *context = priv;
 	context->error = error;
-
-	BUG_ON(!wake_up_process(context->finish_thread));
+	if (!IS_ERR(context->finish_thread)
+			&& (context->finish_thread->state != -1))
```

Comment:
Isn't this situation already handled by the line below?

Reply:
yes